### DISCUSSION TITLE
DAI 2024-2025 - Practical work 3 - Lestiboudois Maxime, Parisod Nathan and Surbeck Léon

### DISCUSSION THREAD

**Discussion by Bananonymous** [heig-vd-dai-course/heig-vd-dai-course#546] [Show and tell]
![Wineventory](https://github.com/La-Kirby-Team/BDR-Project/blob/main/images/Logo.png?raw=true)

[@MaximeLesti](https://github.com/MaximeLesti)
[@Nat004499](https://github.com/Nat004499)
[@Bananonymous](https://github.com/Bananonymous)

[Repo](https://github.com/La-Kirby-Team/BDR-Project)

The **Wineventory** project is a web-based application designed to manage a complex inventory of wines, beers, spirits, non-alcoholic beverages, aperitifs, and related equipment. Leveraging a robust PostgreSQL database and a fully manual SQL query approach, the application ensures efficient and transparent data handling, without relying on ORMs. Users can add, remove, or update products through a user-friendly interface, with real-time stock tracking and alerts for low inventory levels.

From a deployment standpoint, the project follows best practices aligned with the DAI course requirements. It will be containerized using Docker, orchestrated via Docker Compose, and exposed to the internet through a secure and scalable infrastructure. A reverse proxy such as Traefik will manage domain-based routing and automatic HTTPS certificate generation, ensuring secure, globally accessible endpoints. Clear documentation will guide setup on a virtual machine, demonstrating steps for SSH access, DNS configuration, and end-to-end deployment, including how to interact with the API using tools like `curl`.

In addition, the project emphasizes cleanliness and clarity: a well-structured README and codebase, explicit build and deployment commands, and a defined API specification. By focusing on concurrency, caching, performance, and proper HTTP usage (GET, POST, PUT/PATCH, DELETE), the solution meets the course criteria. The end result is a practical, maintainable, and easily extensible web application that aligns with DAI’s learning objectives.

**Comment by ludelafo** [comment]
OK for us but we are not sure to understand the explicit features you will implement in the context of the DAI course (was this description made with ChatGPT? :)).

**Comment by MaximeLesti** [comment]
# Rendu du travail pratique 3: **Winventory** 

**Lien du repo :** [Winventory](https://github.com/La-Kirby-Team/BDR-Project)
**Hash du commit :** [47faf8b](https://github.com/La-Kirby-Team/BDR-Project/commit/47faf8b46f42471f98edec85266bf6efc0f19eb4)

**Reply by ludelafo** [reply]
Bonjour @MaximeLesti,

Je n'ai pas été capable de trouver l'URL où vous avez déployé votre projet sur le dépôt Git. Pouvez-vous me l'indiquer ici pour que je puisse y accéder ?

Merci d'avance,
Ludovic

**Reply by Bananonymous** [reply]
Bonjour, vraiment navré de cette omission, voici :
[winventory.duckdns.org](https://winventory.duckdns.org)
si pour certaines raisons cela ne devait pas fonctionner, l'ip publique est 52.166.217.83

**Reply by ludelafo** [reply]
Merci bien !
